From 9fdef3812e1d933cbaa3f5bdd8d0bddd5e431719 Mon Sep 17 00:00:00 2001
From: Minal Upadhye <minalsunilu@ami.com>
Date: Tue, 8 Sep 2026 12:07:20 +0000
Subject: [PATCH] pwm max min value range recreate

Recreate patch in git format-patch style
Original patch used diff format

Signed-off-by: Minal Upadhye <minalsunilu@ami.com>
Upstream-Status: Inappropriate [AMI specific change]
---
 pwmtachtool.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/pwmtachtool.c b/pwmtachtool.c
index 693f498..8a11198 100644
--- a/pwmtachtool.c
+++ b/pwmtachtool.c
@@ -93,7 +93,7 @@ static int process_arguments( int argc, char **argv,
 		}
 		*fan_num = (unsigned char)strtol( argv[ ++i ], NULL, 10);
 		*rpm_value = (unsigned int)strtol( argv[ ++i ], NULL, 10);
-		action = SET_PWM_DUTYCYCLE;
+                 action = SET_PWM_DUTYCYCLE;
 	}
 	else if( strcmp( argv[ i ], "--set-pwm-dutycycle-value" ) == 0 )
 	{
@@ -104,8 +104,15 @@ static int process_arguments( int argc, char **argv,
 		}
 		*fan_num = (unsigned char)strtol( argv[ ++i ], NULL, 10);
 		*rpm_value = (unsigned int)strtol( argv[ ++i ], NULL, 10);
+		if(*rpm_value < 0 || *rpm_value > 255)
+                {
+                        printf("set-pwm-dutycycle-value should be 0 to 255 \n");
+			return -1;
+		}
 		action = SET_PWM_DUTYCYCLE_VALUE;
+
 	}
+
 	else if( strcmp( argv[i], "--get-pwm-dutycycle" ) == 0)
 	{
 		if (argc < 4)
-- 
2.34.1

